open_seq2seq/encoders/w2l_encoder.py

```diff
@@ -0,0 +1,157 @@
+            "kernel_size": [7], "stride": [1],
+            "num_channels": 250, "padding": "SAME"
+          },
+          {
+            "type": "conv1d", "repeat" : 3,
+            "kernel_size": [11], "stride": [1],
+            "num_channels": 500, "padding": "SAME"
+          },
+          {
+            "type": "conv1d", "repeat" : 1,
+            "kernel_size": [32], "stride": [1],
+            "num_channels": 1000, "padding": "SAME"
+          },
+          {
+            "type": "conv1d", "repeat" : 1,
+            "kernel_size": [1], "stride": [1],
+            "num_channels": 1000, "padding": "SAME" 
+          },
+        ]
+    * **activation_fn** --- activation function to use.
+    * **data_format** (string) --- could be either "channels_first" or
+      "channels_last". Defaults to "channels_last".
+    * **bn_momentum** (float) --- momentum for batch norm. Defaults to 0.99.
+    * **bn_epsilon** (float) --- epsilon for batch norm. Defaults to 1e-3.
+    """
+    super(Wave2LetterEncoder, self).__init__(params, model, name, mode)
+
+  def _get_layer(self, layer_type):
+    if layer_type == "conv1d":
+      return "conv", tf.layers.conv1d
+    elif layer_type == "conv2d":
+      return "conv", tf.layers.conv2d
+
+  def _encode(self, input_dict):
+    """Creates TensorFlow graph for Wav2Letter like encoder.
+
+    Args:
+      input_dict (dict): input dictionary that has to contain
+          the following fields::
+            input_dict = {
+              "source_tensors": [
+                src_sequence (shape=[batch_size, sequence length, num features]),
+                src_length (shape=[batch_size])
+              ]
+            }
+
+    Returns:
+      dict: dictionary with the following tensors::
+
+        {
+          'outputs': hidden state, shape=[batch_size, sequence length, n_hidden]
+          'src_length': tensor, shape=[batch_size]
+        }
+    """
+
+    source_sequence, src_length = input_dict['source_tensors']
+
+    training = (self._mode == "train")
+    dropout_keep_prob = self.params['dropout_keep_prob'] if training else 1.0
+    regularizer = self.params.get('regularizer', None)
+    data_format = self.params.get('data_format', 'channels_last')
+    bn_momentum = self.params.get('bn_momentum', 0.99)
+    bn_epsilon = self.params.get('bn_epsilon', 1e-3)
+
+    conv_inputs = source_sequence
+    batch_size = conv_inputs.get_shape().as_list()[0]
+    if data_format == 'channels_last':
+      conv_feats = conv_inputs #B T F
+    else:
+      conv_feats = tf.transpose(conv_inputs, [0, 2, 1]) #B F T
+
+    # ----- Convolutional layers -----------------------------------------------
+    convnet_layers = self.params['convnet_layers']
+
+    for idx_convnet in range(len(convnet_layers)):
+      layer_type = convnet_layers[idx_convnet]['type']
+      layer_repeat_fixed = convnet_layers[idx_convnet]['repeat']
+      layer_repeat_moving = layer_repeat_fixed
+
+      while(layer_repeat_moving != 0):
```

Comment:
Just a codestyle thing: the brackets around while statement are not necessary, should just be:
while layer_repeat_moving != 0: